hpc_launcher/torch/torchrun_hpc_trampoline.py

```diff
@@ -91,14 +91,24 @@ def main():
                         rank, world_size, MPI.Get_library_version()
                     )
                 )
+
+    # If the world size is only 1, torch distributed doesn't have to be initialized
+    # however, the called application may try to setup torch distributed -- provide env variables
+    # Additionally, some codes (e.g. Huggingface accelerate) will look for these fields
+    os.environ["WORLD_SIZE"] = f"{world_size}"
+    if os.getenv("TORCHRUN_HPC_MASTER_ADDR"):
+        os.environ["MASTER_ADDR"] = os.getenv("TORCHRUN_HPC_MASTER_ADDR")
+    else:
+        # If the mpi rendezvous protocol is set, this should be necessary but some packages still look for it
+        os.environ["MASTER_ADDR"] = "127.0.0.1"
+
+    if os.getenv("TORCHRUN_HPC_MASTER_PORT"):
+        os.environ["MASTER_PORT"] = os.getenv("TORCHRUN_HPC_MASTER_PORT")
     else:
-        # If the world size is only 1, torch distributed doesn't have to be initialized
-        # however, the called application may try to setup torch distributed -- provide env variables
-        os.environ["WORLD_SIZE"] = f"{world_size}"
-        if os.getenv("TORCHRUN_HPC_MASTER_ADDR"):
-            os.environ["MASTER_ADDR"] = os.getenv("TORCHRUN_HPC_MASTER_ADDR")
-        if os.getenv("TORCHRUN_HPC_MASTER_PORT"):
-            os.environ["MASTER_PORT"] = os.getenv("TORCHRUN_HPC_MASTER_PORT")
+        # If the mpi rendezvous protocol is set, this should be necessary but some packages still look for it
+        os.environ["MASTER_ADDR"] = "23456"
+
+    os.environ["LOCAL_RANK"] = f"{local_rank}"
```

Comment:
Doesn't this interfere with `{ROCR,CUDA}_VISIBLE_DEVICES`?

Reply:
I don't think that it does.